account/src/main/java/org/killbill/billing/account/api/DefaultImmutableAccountData.java

```diff
@@ -59,4 +64,15 @@ public Currency getCurrency() {
     public DateTimeZone getTimeZone() {
         return dateTimeZone;
     }
+
+    @Override
+    public UUID getParentAccountId() {
+        return parentAccountId;
```

Comment:
Why the change in `ImmutableAccountData`? This will prevent any re-parenting in the future.